xrpl/models/ledger_objects/ledger_object.py

```diff
@@ -0,0 +1,189 @@
+"""A model for Ledger Objects"""
+
+from __future__ import annotations
+
+from dataclasses import dataclass
+from typing import Any, Dict, Optional, Type, TypeVar
+
+from xrpl.models.base_model import BaseModel
+from xrpl.models.exceptions import XRPLModelException
+from xrpl.models.ledger_objects.ledger_entry_type import LedgerEntryType
+from xrpl.models.required import REQUIRED
+from xrpl.models.utils import require_kwargs_on_init
+
+L = TypeVar("L", bound="LedgerObject")
+
+
+@require_kwargs_on_init
+@dataclass(frozen=True)
+class LedgerObject(BaseModel):
+    """The base model for a Ledger Object."""
+
+    index: Optional[str] = None
+    """
+    The unique ID for this ledger entry; either `index` or `LedgerIndex` is used. In
+    JSON, this field is represented with different names depending on the context and
+    API method. (Note, even though this is specified as "optional" in the code, every
+    ledger entry should have one unless it's legacy data from very early in the XRP
+    Ledger's history.)
+    """
+
+    ledger_index: Optional[str] = None
+    """
+    The unique ID for this ledger entry; either `index` or `LedgerIndex` is used. In
+    JSON, this field is represented with different names depending on the context and
+    API method. (Note, even though this is specified as "optional" in the code, every
+    ledger entry should have one unless it's legacy data from very early in the XRP
+    Ledger's history.)
+    """
+
+    ledger_entry_type: LedgerEntryType = REQUIRED  # type: ignore
+    """
+    The type of ledger entry. Valid ledger entry types include AccountRoot, Offer,
+    RippleState, and others. This field is required.
+    """
+
+    flags: int = 0
+    """
+    Set of bit-flags for this ledger entry.
+    """
+
+    @classmethod
+    def from_dict(cls: Type[L], value: Dict[str, Any]) -> L:
+        """Derive the model from a dict.
+
+        Args:
+            value: The dictionary to derive from.
+
+        Returns:
+            L: The Ledger Object.
+
+        Raises:
+            XRPLModelException: If there is no Ledger Object type is
+                provided or the type mismatches the constructor type.
+        """
+        if cls.__name__ == "LedgerObject":
+            if "ledger_entry_type" not in value:
```

Comment:
Why are you not verifying this condition in the upper level `else` (last branch)?
Shouldn't we throw an exception here: `if cls.__name__ != "LedgerObject" and cls.__name__ not in ["FinalFields", "PreviousFields", "NewFields"] and "ledger_entry_type" not in value` ?

Reply:
As a follow-up question: I notice that 21 unit tests in `test_ledger_object.py` do not have a `LedgerObjectEntry` in their expected json. Isn't that an issue? Shouldn't all necessarily `LedgerObjects` contain this field?